packages/svelte-ux/src/lib/utils/array.ts

```diff
@@ -38,14 +38,19 @@ export function sum(items: (object | null)[], prop?: PropAccessorArg) {
 /**
  * Sum array of objects by property
  */
-export function sumObjects(items: (object | null)[]) {
-  return fromEntries(
-    rollup(
-      items.flatMap((x) => entries(x ?? {})),
-      (values) => sum(values, (d) => (Number.isFinite(d[1]) ? d[1] : 0)),
-      (d) => d[0]
-    )
+export function sumObjects(items: (object | null)[], prop?: PropAccessorArg) {
+  const getProp = propAccessor(prop);
+
+  const result = rollup(
+    items.flatMap((x) => entries(x ?? {})),
+    (values) => sum(values, (d) => {
+      const value = Number(getProp(d[1]));
```

Comment:
I wonder if this should also get the `key` as the second argument (`d[0]` and maybe the whole object as the third (`d`), similar to [just-map-values](https://github.com/angus-c/just?tab=readme-ov-file#just-map-values)